setup.cfg

```diff
@@ -6,7 +6,7 @@ test = pytest
 # using yield with parametrize
 addopts =
     --doctest-modules
-    --disable-pytest-warnings
+#    --disable-pytest-warnings
```

Comment:
what's this?

Reply:
Sorry, I committed that temporarily because I need to see any warnings generated by the tests (to check I captured all deprecation warnings). The problem with this flag is that it hides any warning, not only the ones coming from pytest.

Reply:
wait and in master we're suppressing them?

Reply:
> wait and in master we're suppressing them?

yep, which is not good IMO, but we first need to fix the pytest warnings before enabling the warnings globally becomes useful (otherwise there are way too many pytest warnings to see the relevant ones). 
I think @rth is working on that

Reply:
Also in general, if we want to remove this hiding of the warnings, we should start catching all warnings that are knowingly raised in the tests (as I am doing in this PR), because looking at the travis log from this PR, that clearly does not happen at the moment (there are many deprecation/future/convergence warnings raised in the tests)

Reply:
I 100% agree with you and when I saw the travis output last week I was a bit shocked ;) Doing this has been one of my constant struggles in creating the releases.

Reply:
I opened #11252 to track fixing `DeprecationWarning`s and `FutureWarning`s in tests.